src/Dnn.PersonaBar.Library/Controllers/PersonaBarUserSettingsController.cs

```diff
@@ -32,57 +32,60 @@ public class PersonaBarUserSettingsController : ServiceLocator<IPersonaBarUserSe
         private const string UserSettingsKey = "UserSettings";
 
         #region IUserSettingsController Implementation
-        public void UpdatePersonaBarUserSettings(UserSettings settings, int userId, int portalId)
+        public void UpdatePersonaBarUserSettings(PersistSettings settings, int userId, int portalId)
         {
             var controller = new PersonalizationController();
             var personalizationInfo = controller.LoadProfile(userId, portalId);
-            FixUserSettingsDates(settings);
             Personalization.SetProfile(personalizationInfo, ContainerName, UserSettingsKey, settings);
             controller.SaveProfile(personalizationInfo);
         }
 
-        public UserSettings GetPersonaBarUserSettings()
+        public PersistSettings GetPersonaBarUserSettings()
         {
-            var settings = (UserSettings) Personalization.GetProfile(ContainerName, UserSettingsKey);
-            FixUserSettingsDates(settings);
-            return settings ?? GetDefaultSettings();
+            var settings = Personalization.GetProfile(ContainerName, UserSettingsKey);
+            PersistSettings persistSettings = null;
+            //sync the old settings data into new version.
+#pragma warning disable 618
+            if (settings is UserSettings)
+            {
+                persistSettings = MigrationUserSettings(settings as UserSettings);
```

Comment:
Maybe instead of performing such sync in code it would be better to do a migration step during an instance upgrade?

Reply:
actually i thought that, but it will cause performance issue if the site has a large number of users. with this convert, the information will save into db after PB expand the panel automatically. so that we doesn't need update all of them, but just update as needed.